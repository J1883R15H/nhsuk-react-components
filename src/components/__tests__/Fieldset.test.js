import React from 'react';
import { shallow } from 'enzyme';
import Fieldset from '../fieldset';

describe('Fieldset', () => {
  it('matches snapshot', () => {
    const shallowMount = shallow(<Fieldset>Test</Fieldset>);
    expect(shallowMount).toMatchSnapshot();
    shallowMount.unmount();
  });
  it('renders as a page title', () => {
    const shallowMount = shallow(
      <Fieldset title="Title" asPageHeading>
        Test
      </Fieldset>
    );
    expect(
      shallowMount.containsMatchingElement(
        <h1 className="nhsuk-fieldset__heading nhsuk-fieldset__legend--l">
          Title
        </h1>
      )
    ).toBeTruthy();
    shallowMount.unmount();
  });
});
