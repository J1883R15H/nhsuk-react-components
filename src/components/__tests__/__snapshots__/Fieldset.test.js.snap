// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fieldset matches snapshot 1`] = `
<fieldset
  aria-describedby=""
  className="nhsuk-fieldset"
  style={Object {}}
>
  <legend
    className="nhsuk-fieldset__legend"
  />
  Test
</fieldset>
`;
